clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -67,7 +68,8 @@ public FutureResponse(AbstractResponse responseBody, boolean disconnected, Reque
     private final Set<String> ready = new HashSet<>();
     private final Map<Node, Long> blackedOut = new HashMap<>();
     private final Queue<ClientRequest> requests = new ArrayDeque<>();
-    private final Queue<ClientResponse> responses = new ArrayDeque<>();
+    // Use concurrent queue for responses so that responses may be updated during poll() from a different thread.
+    private final Queue<ClientResponse> responses = new ConcurrentLinkedDeque<>();
```

Comment:
Is this the only field that needs this? Or should all of them be made thread-safe?